eclass/zig-toolchain.eclass

```diff
@@ -0,0 +1,382 @@
+# variable.
+#
+# See ZIG_CPU description for more information.
+zig-toolchain_get_cpu() {
+	if [[ ${#} -ne 1 ]]; then
+		die "${FUNCNAME[0]}: expected 1 arguments, got ${#}"
+	fi
+	local c_target=${1}
+	local c_target_prefix=${c_target%%-*}
+
+	local base_cpu features=""
+
+	case ${c_target_prefix} in
+		i?86)					base_cpu=${c_target_prefix};;
+		loongarch64)			base_cpu=generic_la64;;
+		powerpc | powerpcle)	base_cpu=ppc;;
+		powerpc64)				base_cpu=ppc64;;
+		powerpc64le)			base_cpu=ppc64le;;
+		riscv32)				base_cpu=generic_rv32;;
+		riscv64)				base_cpu=generic_rv64;;
+		*)						base_cpu=generic;;
+	esac
+
+	case ${c_target_prefix} in
+		armv5tel)	features+="+v5te";;
+		armv*)		features+="+${c_target_prefix##armv}";;
+	esac
+
+	case ${c_target_prefix} in
+		arm64)	;;
+		arm*)
+			local is_softfloat=$(CTARGET=${c_target} tc-tuple-is-softfloat)
+			case ${is_softfloat} in
+				only | yes) features+="+soft_float";;
+				softfp | no) features+="-soft_float";;
+				*) die "${FUNCNAME[0]}: tc-tuple-is-softfloat returned unexpected value"
+			esac
+		;;
+	esac
+
+	echo ${base_cpu}${features}
+}
+
+# @FUNCTION: zig-toolchain_find_installation
+# @DESCRIPTION:
+# Detects suitable Zig installation and sets ZIG_VER and ZIG_EXE
+# variables.
+#
+# See ZIG_EXE and ZIG_VER descriptions for more information.
+zig-toolchain_find_installation() {
+	# Adapted from https://github.com/gentoo/gentoo/pull/28986
+	# Many thanks to Florian Schmaus (Flowdalic)!
+
+	[[ -n ${ZIG_SLOT} ]] || die "${FUNCNAME[0]}: ZIG_SLOT must be set"
+	if ver_test "${ZIG_SLOT}" -lt "0.13"; then
+		die "${ECLASS}: ZIG_SLOT must be >= 0.13, found ${ZIG_SLOT}"
+	fi
+
+	einfo "Searching Zig ${ZIG_SLOT}..."
+
+	local zig_supported_versions=(
+		"9999"
+		"0.13.1"
+		"0.13.0"
+	)
+
+	local base_path="${BROOT}/usr/bin"
+
+	local selected_path selected_ver
+	for selected_ver in "${zig_supported_versions[@]}"; do
+		# Check if candidate satisfies ZIG_SLOT condition.
+		local candidate_slot
+		candidate_slot=$(ver_cut 1-2 ${selected_ver})
+		if ver_test "${candidate_slot}" -ne ${ZIG_SLOT}; then
+			continue
+		fi
+
+		local candidate_path
+		# Prefer "dev-lang/zig" over "dev-lang/zig-bin"
+		candidate_path="${base_path}/zig-${selected_ver}"
+		if [[ -x "${candidate_path}" ]]; then
+			selected_path="${candidate_path}"
+			break;
+		fi
+
+		candidate_path="${base_path}/zig-bin-${selected_ver}"
+		if [[ -x "${candidate_path}" ]]; then
+			selected_path="${candidate_path}"
+			break;
+		fi
+	done
+
+	if [[ -z "${selected_path}" ]]; then
+		die "Could not find (suitable) Zig at \"${base_path}\""
+	fi
+
+	export ZIG_EXE="${selected_path}"
+	export ZIG_VER="${selected_ver}"
+	# Sanity check, comment from upstream:
+	# > Check libc++ linkage to make sure Zig was built correctly,
+	# > but only for "env" and "version" to avoid affecting the
+	# > startup time for build-critical commands
+	# > (check takes about ~10 μs)
+	"${ZIG_EXE}" version > /dev/null \
+		|| die "Sanity check failed for \"${ZIG_EXE}\""
```

Comment:
```suggestion
	"${ZIG_EXE}" version > /dev/null ||
		die "Sanity check failed for \"${ZIG_EXE}\""
```

Don't use backslashes when they can be easily avoided.

Reply:
Will do so, thanks

Reply:
16cdbcf62ca8...d45c79cffb21